runtime/registry.go

```diff
@@ -203,15 +215,17 @@ func (r *Registry) Query(q *query.Query, local, internal bool) (*iterator.Iterat
 
 			for _, r := range records {
 				r.Lock()
-				if q.MatchesKey(r.DatabaseKey()) ||
-					!r.Meta().CheckValidity() ||
-					!r.Meta().CheckPermission(local, internal) ||
-					!q.MatchesRecord(r) {
+				var (
+					matchesKey    = q.MatchesKey(r.DatabaseKey())
+					isValid       = r.Meta().CheckValidity()
+					isAllowed     = r.Meta().CheckPermission(local, internal)
+					matchesRecord = q.MatchesRecord(r)
```

Comment:
MatchesRecord is far more expensive than the other ones, so far that it might be worth doing an extra lock for it. What do you think?

Reply:
Good point. will change that